devtools/cli/src/main/java/io/quarkus/cli/Add.java

```diff
@@ -0,0 +1,53 @@
+package io.quarkus.cli;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+import io.quarkus.devtools.commands.AddExtensions;
+import io.quarkus.devtools.commands.data.QuarkusCommandOutcome;
+import io.quarkus.devtools.project.QuarkusProject;
+import io.quarkus.platform.tools.config.QuarkusPlatformConfig;
+import picocli.CommandLine;
+
+@CommandLine.Command(name = "add", mixinStandardHelpOptions = false, description = "Add extension(s) to current project.")
+public class Add implements Callable<Integer> {
+
+    @CommandLine.Option(names = { "-h", "--help" }, usageHelp = true, description = "display this help message")
+    boolean usageHelpRequested;
+
+    @CommandLine.Parameters(arity = "1", paramLabel = "EXTENSION", description = "extension to add to project")
+    Set<String> extensions;
+
+    @CommandLine.ParentCommand
+    QuarkusCli parent;
+
+    @CommandLine.Spec
+    CommandLine.Model.CommandSpec spec;
+
+    @Override
+    public Integer call() {
+        Path projectDirectory = Paths.get(System.getProperty("user.dir"));
+        try {
+            QuarkusProject quarkusProject = QuarkusProject.resolveExistingProject(projectDirectory,
```

Comment:
@aloubyansky @maxandersen @patriot1burke , this command is not going to be compatible with gradle. This is why for the add/remove/list extension, I would delegate the call to the wrapper plugin to make this command compatible on both mvn and gradle. If there is no wrapper, then you could fallback to this and if it target a gradle project fail with instruction.

Reply:
And even more now that I see you already implemented an `ExecuteUtil` :)

Reply:
Can you explain why it won't work?

Reply:
sure @patriot1burke, because Gradle API is not available outside of a gradle plugin, without it, we are not able to properly list extensions (neither add and remove which rely on the list): https://github.com/quarkusio/quarkus/blob/master/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/GenericGradleBuildFile.java

Reply:
can confirm it doesn't work:

```
 quarkus add hibernate-orm-panache
Unable to add extension: This feature is not yet implemented outside of the Gradle Plugin.
```